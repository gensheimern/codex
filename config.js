module.exports = {
	DB_HOST: process.env.DB_HOST || '159.65.115.221',
	DB_USER: process.env.DB_USER || 'root',
	DB_PASS: process.env.DB_PASS || 'teamcodex',
	DB_NAME: process.env.DB_NAME || 'lunch_planner',
	JWT_SECRET: process.env.JWT_SECRET || 'secret',
	MAIL_HOST: process.env.MAIL_HOST || '159.65.115.221',
	MAIL_USER: process.env.MAIL_USER || 'support@codex-team.de',
	MAIL_PASS: process.env.MAIL_PASS || 'teamcodex123',
};
