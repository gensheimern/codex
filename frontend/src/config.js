const config = {
	debug: process.env.REACT_APP_DEBUG === 'TRUE',
	apiPath: process.env.REACT_APP_API_PATH,
	wsPath: process.env.REACT_APP_WS_PATH,
	basePath: process.env.REACT_APP_BASE_PATH,
}

export default config;