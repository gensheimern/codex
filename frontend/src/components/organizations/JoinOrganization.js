import React from 'react';
import RaisedButton from 'material-ui/RaisedButton';
import TextField from 'material-ui/TextField'
import config from '../../config';

export default class JoinOrganization extends React.Component {
	constructor(props) {
		super(props);
	
		this.state = {
			password: '',
			retypePassword: '',
			changedError: false,
		};

		this.joinOrganization = this.joinOrganization.bind(this);
	}

	joinOrganization(organizationId, password) {
		this.setState({
			changedError: false,
		});

		fetch(`${config.apiPath }/user/me/organizations/${organizationId}`, {
            method: 'POST',
            headers: {
				'Content-Type': 'application/json',
				'X-Access-Token': localStorage.getItem('apiToken'),
			},
			body: JSON.stringify({
				password,
			}),
		})
		.then((res) => {
            if(!res.ok) {
                throw new Error("Response not ok.");
            } else if(res.status !== 200) {
                throw new Error("An error occured.");
            }
            return res.json();
		})
		.then((res) => {
            this.setState({
				changed: true,
				showOtherOrganizations: false,
				changedError: false,
			});

			this.props.saveChange(organizationId);
			this.props.close();
        }).catch((error) => {
			this.setState({
				changedError: true,
			});
        });
	}

	createOrganization() {
		console.log('Created new Organization.');
		// TODO:
	}

	handleChange = name => e => {
		this.setState({
			[name]: e.target.value
		});
	}

	render() {
		if (!this.props.show) {
			return null;
		}

		const errorMessage = (
			<p style={{
				color: 'red',
			}}>Incorrect Password.</p>
		);

		return (
			<React.Fragment>
				<h4>Join '{this.props.organization.name}':</h4>
				<TextField
					type="password"
					floatingLabelText="Organization password"
					value={this.state.password}
					onChange={this.handleChange('password')}
				/>
				<RaisedButton
					label="Join"
					primary={true}
					onClick={() => {
						this.joinOrganization(this.props.id, this.state.password);
					}}
					style={{
						float: 'right',
						marginTop: '25px',
					}}
				/>
				{this.state.changedError ? errorMessage : null}
			</React.Fragment>
		);
	}
}
