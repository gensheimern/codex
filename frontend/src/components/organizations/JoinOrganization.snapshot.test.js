/**
 * @author:Stella Neser
 *  Komponente ist ein Snapshottest
 */
import React from 'react';
import renderer from'react-test-renderer';
import JoinOrganization from './JoinOrganization';
import config from '../../config';
import {MemoryRouter} from 'react-router-dom';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
const fetchMock = require('fetch-mock');

describe('JoinOrganization Snapshot', () => {
    test('renders', () =>{
        const component = renderer.create(
            <MemoryRouter>
                <MuiThemeProvider>
                    <JoinOrganization/>
                </MuiThemeProvider>
            </MemoryRouter>
        );
        let tree= component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});