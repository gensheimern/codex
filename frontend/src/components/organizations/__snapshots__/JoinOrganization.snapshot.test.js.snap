// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JoinOrganization Snapshot renders 1`] = `null`;
